lib/routes/api/accounts.js

```diff
@@ -213,15 +217,48 @@ async function validateCreateCall(logger, sid, req) {
   }
   else {
     delete obj.application_sid;
+    // Fetch Speech provider credential From DB
+    let speech_creds = await SpeechCredential.retrieveAll(req.user.account_sid);
+    if (!speech_creds || speech_creds.length === 0) {
+      // try to fetch speech cred from service provider
+      const {lookupAccountBySid} = req.app.locals;
+      const account = await lookupAccountBySid(req.user.account_sid);
+      speech_creds = await SpeechCredential.retrieveAllForSP(account.service_provider_sid);
+    }
 
-    // TODO: these should be retrieved from account, using account_sid if provided
-    Object.assign(obj, {
-      speech_synthesis_vendor: 'google',
-      speech_synthesis_voice: 'en-US-Wavenet-C',
-      speech_synthesis_language: 'en-US',
-      speech_recognizer_vendor: 'google',
-      speech_recognizer_language: 'en-US'
+    if (!speech_creds || speech_creds.length === 0)
+      throw new DbErrorBadRequest(`missing speech synthesizer for account ${sid}`);
```

Comment:
Please change to "missing speech credentials.."

Reply:
remove DB query as below comment does not need this action